exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/decoder/Packet.java

```diff
@@ -319,7 +324,11 @@ public int getDst_port() {
     byte[] data = null;
     if (packetLength >= payloadDataStart) {
       data = new byte[packetLength - payloadDataStart];
-      System.arraycopy(raw, ipOffset + payloadDataStart, data, 0, data.length);
+      try {
+        System.arraycopy(raw, ipOffset + payloadDataStart, data, 0, data.length);
```

Comment:
Maybe you have to set isCorrupt = false;
After first corrupted row, the following lines are always iCorrupt = true;

Reply:
I looked through this, and found that the way the pcap plugin was handling boolean fields needed some work.  So, now all the tcp flags and the `is_corrupt` fields are actual booleans.  
I included an additional unit test to demonstrate this.